src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,246 @@
+
+        try {
+            loadSavedText(defaultPath, listOfItems);
+            System.out.println("Previous entries uploaded.");
+            System.out.println(separatingLine);
+        }catch (FileNotFoundException e){
+           File f =new File(defaultPath);
+           f.createNewFile();
+           System.out.println("Save not detected, creating new save file.");
+           System.out.println(separatingLine);
+        }
+        listWordCount = listOfItems.size();
+
+        //loop begins:
+        while (!hasEnded) {
+            Scanner in = new Scanner(System.in);
+            inputStatement = in.nextLine();
+            if (inputStatement.equalsIgnoreCase("bye")) {
+                //signals end of programme
+                hasEnded = true;
+            } else if (inputStatement.equalsIgnoreCase("list") ){
+                System.out.println(separatingLine);
+                //print out list of items
+                try {
+                    printArrangedList(listOfItems);
+                } catch (IllegalCommandsException e){
+                    //throws exception when list is empty.
+                    System.out.println("Your list is empty, no lollygagging");
+                }
+                System.out.println(separatingLine);
+            } else if(inputStatement.startsWith("done")){
+                System.out.println(separatingLine);
+                try {
+                    markAsDone(inputStatement, listOfItems);
+                } catch (IllegalCommandsException e){
+                    //throws exception when doing task that is already done.
+                    System.out.println("Calm done mate, you finished that already.");
+                }catch (NumberFormatException e){
+                    //throws exception when done command is not followed by a number.
+                    System.out.println("Tip of the day: End your done command with a number next time");
+                }catch (ArrayIndexOutOfBoundsException | NullPointerException e){
+                    //throws exception when done command is followed by an empty task or task outside of array.
+                    System.out.println("You can't do something that doesn't exist, or can you?");
+                }
+                System.out.println(separatingLine);
+            } else if(inputStatement.startsWith("delete")){
+                System.out.println(separatingLine);
+                try {
+                    deleteItem(inputStatement, listOfItems);
+                    listWordCount = listWordCount - 1;
+                    System.out.println("Now you have " + (listWordCount) + " tasks in the list.");
+                }catch(IndexOutOfBoundsException|NumberFormatException e){
+                    System.out.println("You can't delete that!");
+                }
+                System.out.println(separatingLine);
+            }else if(inputStatement.startsWith("todo")|inputStatement.startsWith("event")
```

Comment:
Maybe avoid complicated expressions? Please refer to: https://nus-cs2113-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#guideline-name-well